nixos/tests/firewall.nix

```diff
@@ -3,14 +3,30 @@
 import ./make-test-python.nix ( { pkgs, nftables, ... } : {
   name = "firewall" + pkgs.lib.optionalString nftables "-nftables";
   meta = with pkgs.lib.maintainers; {
-    maintainers = [ eelco ];
+    maintainers = [ eelco rvfg ];
   };
 
   nodes =
     { walled =
         { ... }:
-        { networking.firewall.enable = true;
-          networking.firewall.logRefusedPackets = true;
+        { networking.firewall = {
+            enable = true;
+            logRefusedPackets = true;
+            allowedTCPPorts = [ 25 993 8005 ];
+            allowedTCPPortRanges = [
+              { from = 980; to = 1000; }
+              { from = 990; to = 1010; }
+              { from = 8000; to = 8010; }
+            ];
+            interfaces.eth0 = {
+              allowedTCPPorts = [ 10003 ];
+              allowedTCPPortRanges = [ { from = 10000; to = 10005; } ];
+            };
+            interfaces.eth3 = {
+              allowedUDPPorts = [ 10003 ];
+              allowedUDPPortRanges = [ { from = 10000; to = 10005; } ];
+            };
+          };
```

Comment:
This part is to test if the syntax of the generated ruleset is correct. The firewall functionality is not tested here.

Reply:
```suggestion
            # Syntax smoke test, not actually verified otherwise
            allowedTCPPorts = [ 25 993 8005 ];
            allowedTCPPortRanges = [
              { from = 980; to = 1000; }
              { from = 990; to = 1010; }
              { from = 8000; to = 8010; }
            ];
            interfaces.eth0 = {
              allowedTCPPorts = [ 10003 ];
              allowedTCPPortRanges = [ { from = 10000; to = 10005; } ];
            };
            interfaces.eth3 = {
              allowedUDPPorts = [ 10003 ];
              allowedUDPPortRanges = [ { from = 10000; to = 10005; } ];
            };
          };
```